xapian-applications/omega/omegatest

```diff
@@ -414,6 +414,20 @@ printf '$highlight{$cgi{text},$cgi{list},$cgi{open},$cgi{close}}' > "$TEST_TEMPL
 testcase 'A list of <b>words</b>' list="words" text="A list of words" open="<b>" close="</b>"
 testcase 'A *list* of *words*' list="words${tab}list" text="A list of words" open="*" close="*"
 
+#Test setting seterror and mset size, should not run the query after setting error.
+printf '$if{$cgi{ERR},$seterror{$cgi{ERR}}}$msize$if{$error,!$error!}' > "$TEST_TEMPLATE"
+testcase '1' P=text
+testcase '0!boo!' P=text ERR=boo
+
+#Test arguments inside seterror are evaluated
+printf '$set{error,sample error}$seterror{$opt{error}}$msize$if{$error,!$error!}' > "$TEST_TEMPLATE"
+testcase '0!sample error!' P=text
+
+#Test msize when error set after running query, should not affect running of query.
+printf '$last$if{$cgi{ERR},$seterror{$cgi{ERR}}}$msize$if{$error,!$error!}' > "$TEST_TEMPLATE"
+testcase '11' P=text
+testcase '11!boo!' P=text ERR=boo
+
```

Comment:
It would be good to test with some HTML-special characters in the error to make sure it doesn't get sent through HTML entity encoding (as the C++ `throw` approach does).  Making the string `<sample error>` in the second case would do that.